onlyoffice-bin: init at 6.1.0

nh2 opened the pull request:
###### Motivation for this change

AGPL based office suite with good compatibility for Microsoft Office documents.

Imported from:

https://github.com/icebox-nix/iceberg/blob/a49fa345b53cdf51a480cb53623a8194fb2d4b56/pkgs/office/onlyoffice-bin/default.nix


### Comaintainers wanted :heart: 

I need others that want to be co-maintainer on this package, please ping me here and I'll add you.


##### Note on fonts:

OnlyOffice does not distribute unfree fonts, but makes it easy to pick up
any fonts you install. See:

* https://helpcenter.onlyoffice.com/en/installation/docs-community-install-fonts-linux.aspx
* https://www.onlyoffice.com/blog/2020/04/how-to-add-new-fonts-to-onlyoffice-desktop-editors/

As recommended there, you should download

    arial.ttf, calibri.ttf, cour.ttf, symbol.ttf, times.ttf, wingding.ttf

into `~/.local/share/fonts/`, otherwise the default template fonts, and
things like bullet points, will not look as expected.

```
TODO: Find out which of these fonts we'd be allowed to distribute along
      with this package, or how to make this easier for users otherwise.
```


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).


I've also checked that it runs on Nix on Ubuntu, works well.

---

CC @LEXUGE @onny @emmanuelrosa @mweinelt @aszlig from https://discourse.nixos.org/t/onlyoffice-desktopeditors-packaged/7226/14

CC @tomberek from https://discourse.nixos.org/t/onlyoffice-on-nixos/4351

nh2 committed:
onlyoffice-bin: init at 6.1.0

Imported from:

https://github.com/icebox-nix/iceberg/blob/a49fa345b53cdf51a480cb53623a8194fb2d4b56/pkgs/office/onlyoffice-bin/default.nix

nh2 commented:
CC also @GTrunSec from https://discourse.nixos.org/t/onlyoffice-desktopeditors-packaged/7226/16?u=nh2

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/onlyoffice-desktopeditors-packaged/7226/17

r-rmcgibbo commented:
Result of `nixpkgs-review pr 116343` at d1a89515 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/1a77c64d7aaf4105289d8eb26ff1a0b9">onlyoffice-bin</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/office/onlyoffice-bin/default.nix:128:3:
```
    |
128 |   installPhase = ''
    |   ^
```
</li>
  </ul>
</details>

nh2 commented:
>     * `installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Done.

SuperSandro2000 commented on pkgs/applications/office/onlyoffice-bin/default.nix:
```diff
@@ -0,0 +1,163 @@
+{ stdenv
+, lib
+, fetchurl
+
+, alsaLib
```
```suggestion
, fetchurl
, alsaLib
```

nh2 replied:
Added a comment to make clearer what the gap was for.

SuperSandro2000 commented on pkgs/applications/office/onlyoffice-bin/default.nix:
```diff
@@ -0,0 +1,163 @@
+  # TODO: Find out which of these fonts we'd be allowed to distribute along
+  #       with this package, or how to make this easier for users otherwise.
+
+  noto-fonts-cjk = fetchurl {
```
Please use the noto-fonts-cjk that are already in nixpkgs. https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/noto-fonts/default.nix#L77

nh2 replied:
That seems to be something slightly different, and 10x much larger:

Nixpkgs' `noto-fonts-cjk`:

```
115M	/nix/store/rylkxlfxqp95z0hxyx6mqzfxvnfpw3ld-noto-fonts-cjk-2.001/share/fonts/opentype/noto-cjk/NotoSansCJK.ttc
```

This one used:

```
16M	/nix/store/zwpnc457ln4nf4brri9m2pcqa678cp48-onlyoffice-desktopeditors-6.1.0/share/desktopeditors/fonts/n1nd1bgihg5gwv2if7kcfqkbrlg8zwv4-NotoSansCJKsc-Regular.otf
```

I suppose `.ttc` is some kind of collection of multiple fonts?

nh2 replied:
@GTrunSec You add fonts here: https://github.com/GTrunSec/onlyoffice-desktopeditors-flake/blob/6a404e6916a39e8b51211d64c438a3b25d139013/flake.nix#L70-L71

@LEXUGE You add fonts here:

https://github.com/icebox-nix/iceberg/blob/a49fa345b53cdf51a480cb53623a8194fb2d4b56/pkgs/office/onlyoffice-bin/default.nix#L67

Can any of you tell whether these fonts are strictly necessary in onlyoffice?

As far as I understand, OnlyOffice should pick up the system fonts and the ones in `~/.local/share/fonts`.

Do you think it would be OK to remove the fonts installation as done here so far, or did you find them to be necessary? Both of them seem to be for non-Latin alphabets, and I'm not really an expert at testing for those.

LEXUGE replied:
The biggest problem with the OnlyOffice is that it does NOT correctly pick up the fonts 
present in the system. For other fonts, primarily CJK fonts, we have to include it. In addition to that, those added fonts are strict on format, and we tried in stock cjk fonts which don't work.

nh2 replied:
@LEXUGE To clarify with "we tried in stock cjk fonts", you mean the `noto-fonts-cjk` package in nixpkgs with its `NotoSansCJK.ttc` file I mentioned above?

LEXUGE replied:
Yes, the tic file in nixpkgs is what we tried.

nh2 replied:
@LEXUGE Thanks, I've added a comment to the derivation about this.

SuperSandro2000 commented on pkgs/applications/office/onlyoffice-bin/default.nix:
```diff
@@ -0,0 +1,163 @@
+  version = "6.1.0";
+  minor = "90";
+  src = fetchurl {
+    url =
+      "https://github.com/ONLYOFFICE/DesktopEditors/releases/download/v${version}/onlyoffice-desktopeditors_${version}-${minor}_amd64.deb";
```
```suggestion
    url = "https://github.com/ONLYOFFICE/DesktopEditors/releases/download/v${version}/onlyoffice-desktopeditors_${version}-${minor}_amd64.deb";
```

nh2 replied:
Done.

SuperSandro2000 commented on pkgs/applications/office/onlyoffice-bin/default.nix:
```diff
@@ -0,0 +1,163 @@
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share
+    mkdir -p $out/{bin,lib}
```
```suggestion
    mkdir -p $out/{share,bin,lib}
```

nh2 replied:
Done.

SuperSandro2000 commented on pkgs/applications/office/onlyoffice-bin/default.nix:
```diff
@@ -0,0 +1,163 @@
+  meta = with lib; {
+    description = "Office suite that combines text, spreadsheet and presentation editors allowing to create, view and edit local documents";
+    homepage = "https://www.onlyoffice.com/";
+    platforms = platforms.linux;
```
I think this is x86_64 only, right?

nh2 replied:
Oh yes, fixed.

nh2 committed:
maintainers: Add gtrunsec

As requested on https://discourse.nixos.org/t/onlyoffice-desktopeditors-packaged/7226/21
